FqnToConfig doesn't skip some modules when set _default.

### Description

When using `FqnToConfig` (or `ModuleFqnToConfig`) with a configuration that includes both:
- A regex pattern matching to `None` (intended to skip quantization for certain modules)
- A `_default` configuration (intended to quantize other modules)

Modules that match the `None` pattern are incorrectly still quantized by `_default` instead of being skipped.

### Steps to Reproduce

```
from collections import OrderedDict
import torch
import torch.nn as nn
from torchao.quantization import (
    quantize_,
    Int8DynamicActivationInt8WeightConfig,
    ModuleFqnToConfig,
)

class TestModel(nn.Module):
    def __init__(self):
        super().__init__()
        self.time_embed = nn.Sequential(
            nn.Linear(1024, 1024),
            nn.ReLU(),
            nn.Linear(1024, 1024),
        )
        self.linear1 = nn.Linear(1024, 1024)

    def forward(self, x):
        x = self.time_embed(x)
        x = self.linear1(x)
        return x

model = TestModel()
model.eval()

# Expected: time_embed should NOT be quantized, linear1 should be quantized
cfg = ModuleFqnToConfig(
    module_fqn_to_config=OrderedDict([
        ("re:.*time_embed.*", None),  # Skip time_embed
        ("_default", Int8DynamicActivationInt8WeightConfig()),  # Quantize other modules
    ])
)

quantize_(model, cfg, filter_fn=None)
```

error logs: 
Traceback (most recent call last):
  File "/root/workspace/optkit/examples/ttt.py", line 43, in <module>
    quantize_(model, cfg, filter_fn=None)
  File "/usr/local/lib/python3.10/dist-packages/torchao/quantization/quant_api.py", line 502, in quantize_
    _fqn_to_config_handler(module, module_fqn, config)
  File "/usr/local/lib/python3.10/dist-packages/torchao/quantization/quant_api.py", line 2690, in _fqn_to_config_handler
    return handler(module, c)
  File "/usr/local/lib/python3.10/dist-packages/torchao/quantization/quant_api.py", line 1629, in _int8_dynamic_activation_int8_weight_transform
    assert hasattr(module, "weight"), (
AssertionError: applying int8 dynamic activation int8 weight quant requires module to have weight attributebut {module} does not have one

--- a/test/quantization/test_quant_api.py
+++ b/test/quantization/test_quant_api.py
@@ -692,6 +692,43 @@ class TestFqnToConfig(TestCase):
         assert str(linear_model).startswith(expected_starting_str)
         assert str(linear_model.linear1.weight) in str(linear_model)
 
+    def test_fqn_to_config_regex_skip(self):
+        """Test that regex pattern with None config skips matching modules."""
+
+        class TestModel(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.time_embed = torch.nn.Sequential(
+                    torch.nn.Linear(128, 128), torch.nn.Linear(128, 128)
+                )
+                self.linear1 = torch.nn.Linear(128, 128)
+
+            def forward(self, x):
+                x = self.time_embed(x)
+                x = self.linear1(x)
+                return x
+
+        model = TestModel().eval()
+
+        cfg = FqnToConfig(
+            {
+                "re:.*time_embed.*": None,
+                "_default": Float8WeightOnlyConfig(),
+            }
+        )
+
+        quantize_(model, cfg, filter_fn=None)
+
+        # time_embed linears should NOT be quantized (regex matched with None)
+        for name, mod in model.time_embed.named_modules():
+            if isinstance(mod, torch.nn.Linear):
+                assert not isinstance(mod.weight, Float8Tensor), (
+                    f"time_embed.{name}.weight should not be quantized"
+                )
+
+        # linear1 should be quantized via _default
+        assert isinstance(model.linear1.weight, Float8Tensor)
+
     def test_quantize_param_fqn_exact(self):
         from transformers import AutoConfig
         from transformers.models.llama4.modeling_llama4 import Llama4TextMoe

--- a/torchao/quantization/quant_api.py
+++ b/torchao/quantization/quant_api.py
@@ -1980,6 +1980,8 @@ def _fqn_to_config_handler(
                 if c is not None:
                     handler = _QUANTIZE_CONFIG_HANDLER[type(c)]
                     return handler(module, c)
+                else:
+                    return module
 
     # If no module_fqn or parameter_fqn matches, then we apply _default
     if not parameter_config_found:

